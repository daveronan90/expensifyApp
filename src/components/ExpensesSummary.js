import numeral from "numeral";
import React from "react";
import { connect } from "react-redux";
import selectExpenses from "../selectors/expenses";
import expensesTotal from "../selectors/expenses-total";

export const ExpensesSummary = ({ expensesCount, expensesTotal }) => {
  const expenseWord = expensesCount === 1 ? "expense" : "expenses";
  const formattedExpensesTotal = numeral(expensesTotal / 100).format(
    "$0,00.00"
  );

  return (
    <div>
      <h1>
        Viewing {expensesCount} {expenseWord} totalling {formattedExpensesTotal}
      </h1>
    </div>
  );
};

const mapStateToProps = ({ expenses, filters }) => {
  const visibileExpenses = selectExpenses(expenses, filters);
  return {
    expensesCount: expenses ? 0 : visibileExpenses.length,
    expensesTotal: expensesTotal(visibileExpenses),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
