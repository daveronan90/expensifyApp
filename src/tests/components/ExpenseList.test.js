import { shallow } from "enzyme";
import React from "react";
import { ExpenseList } from "../../components/ExpenseList";
import expenses from "../fixtures/expenses";

test("Should render ExpenseList with expenses", () => {
  const wrapper = shallow(<ExpenseList expenses={expenses} />);

  expect(wrapper).toMatchSnapshot();
});

test("Should render ExpenseList without expenses and message", () => {
  const wrapper = shallow(<ExpenseList expenses={[]} />);

  expect(wrapper).toMatchSnapshot();
});
