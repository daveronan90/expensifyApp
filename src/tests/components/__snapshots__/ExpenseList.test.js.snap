// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseList with expenses 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <ExpenseListItem
      amount={195}
      createdAt={0}
      description="gum"
      id="1"
      key="1"
      note=""
    />
    <ExpenseListItem
      amount={2000}
      createdAt={-345600000}
      description="gas"
      id="2"
      key="2"
      note=""
    />
    <ExpenseListItem
      amount={20000}
      createdAt={345600000}
      description="rent"
      id="3"
      key="3"
      note=""
    />
  </div>
</div>
`;

exports[`Should render ExpenseList without expenses and message 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <div
      className="list-item list-item--message"
    >
      <span>
        No expenses
      </span>
    </div>
  </div>
</div>
`;
